### ISSUE TITLE
Rule Change: Add suggestions to use-isnan

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#17978]
### What rule do you want to change?

use-isnan

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
value == NaN
```

### What does the rule currently do for this code?

Report an error - and nothing else.

### What will the rule do after it's changed?

Also provide a suggestion to switch to `isNaN(value)`.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Either `Number.isNaN` or `isNaN` could be suggested. Maybe the docs should state which is preferred and why?

**Comment by nzakas** [comment]
Sounds good to me. I think the intent of adding `Number.isNaN()` was to encourage people to move away from `isNaN()`, so I'd be fine with recommending the use of `Number.isNaN()` in the docs as the preferred approach.

**Comment by mdjermanovic** [comment]
https://github.com/eslint/eslint/commit/2d11d46e890a9f1b5f639b8ee034ffa9bd453e42 adds suggestions for cases where `NaN` is used with  `==`, `!=`, `===`, or `!==`. Are there any other cases for which we could implement suggestions, or can we close this issue now?

**Comment by nzakas** [comment]
I think it's safe to close. :+1:

**Comment by StyleShit** [comment]
I think it should be reopened because `indexOf` and `lastIndexOf` can also be fixed. That's why I noted in the PR that it partially handles this issue and not closing it
